src/test/compile-fail/pub-restricted-2.rs

```diff
@@ -0,0 +1,31 @@
+// Copyright 2016 The Rust Project Developers. See the COPYRIGHT
+// file at the top-level directory of this distribution and at
```

Comment:
Meta note: Let's make a subdirectory for these tests.

Perhaps `src/test/compile-fail/privacy`? If we want to further designate tests aimed at the `pub(restricted)` RFC, then perhaps `src/test/compile-fail/privacy/restricted`?